miden-crypto/src/merkle/mmr/partial.rs

```diff
@@ -210,8 +212,8 @@ impl PartialMmr {
     ///
     /// When `track` is `true` the new leaf is tracked.
     pub fn add(&mut self, leaf: RpoDigest, track: bool) -> Vec<(InOrderIndex, RpoDigest)> {
-        self.forest += 1;
-        let merges = self.forest.trailing_zeros() as usize;
+        self.forest.increment();
+        let merges = self.forest.smallest_tree_height();
         let mut new_nodes = Vec::with_capacity(merges);
 
         let peak = if merges == 0 {
```

Comment:
Below on line 227 we use `forest_to_rightmost_index()` function. Should it also be a method on the `MountainRange` struct?

Reply:
Moved to `MountainRange::rightmost_in_order_index` and `MountainRange::root_in_order_index`. I didn't make any higher level changes, see my reasoning in `merkle::mmr::full`.